flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java

```diff
@@ -0,0 +1,361 @@
+
+    Optional<org.apache.iceberg.expressions.Expression> actual =
+        FlinkFilters.convert(resolve(Expressions.$("field1").isLessOrEqual(Expressions.lit(1))));
+    Assert.assertTrue(actual.isPresent());
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual.get());
+
+    Optional<org.apache.iceberg.expressions.Expression> actual1 =
+        FlinkFilters.convert(resolve(Expressions.lit(1).isGreaterOrEqual(Expressions.$("field1"))));
+    Assert.assertTrue(actual1.isPresent());
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual1.get());
+  }
+
+  @Test
+  public void testIsNull() {
+    Expression expr = resolve(Expressions.$("field1").isNull());
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    UnboundPredicate<Object> expected = org.apache.iceberg.expressions.Expressions.isNull("field1");
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual.get());
+  }
+
+  @Test
+  public void testIsNotNull() {
+    Expression expr = resolve(Expressions.$("field1").isNotNull());
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    UnboundPredicate<Object> expected = org.apache.iceberg.expressions.Expressions.notNull("field1");
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual.get());
+  }
+
+  @Test
+  public void testAnd() {
+    Expression expr = resolve(
+        Expressions.$("field1").isEqual(Expressions.lit(1)).and(Expressions.$("field2").isEqual(Expressions.lit(2L))));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    And and = (And) actual.get();
+    And expected = (And) org.apache.iceberg.expressions.Expressions.and(
+        org.apache.iceberg.expressions.Expressions.equal("field1", 1),
+        org.apache.iceberg.expressions.Expressions.equal("field2", 2L));
+
+    Assert.assertEquals(expected.op(), and.op());
+    Assert.assertEquals(expected.left().op(), and.left().op());
+    Assert.assertEquals(expected.right().op(), and.right().op());
+  }
+
+  @Test
+  public void testOr() {
+    Expression expr = resolve(
+        Expressions.$("field1").isEqual(Expressions.lit(1)).or(Expressions.$("field2").isEqual(Expressions.lit(2L))));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    Or or = (Or) actual.get();
+    Or expected = (Or) org.apache.iceberg.expressions.Expressions.or(
+        org.apache.iceberg.expressions.Expressions.equal("field1", 1),
+        org.apache.iceberg.expressions.Expressions.equal("field2", 2L));
+
+    Assert.assertEquals(expected.op(), or.op());
+    Assert.assertEquals(expected.left().op(), or.left().op());
```

Comment:
I think these should use `assertPredicatesMatch` instead of just checking the operation.